src/lib/device_tree/src/lib.rs

```diff
@@ -92,10 +137,10 @@ fn align4(x: usize) -> usize {
     (x + 3) & !3
 }
 
-/// In-memory device tree traversal.
-/// Does not perform any sanity checks.
 impl<'a> FdtReader<'a> {
     pub fn new(drv: &'a dyn Driver) -> Result<FdtReader<'a>> {
+        // TODO: Instead of reading each number one by one consider
+        // reading the whole size of the header at once, and then just
```

Comment:
and then just ... ?

Reply:
I am very sorry, somehow changes in this PR got some unrelated changes in this file, reverted all of them to prevent the conflict with the other PR.